lightning/src/ln/channelmanager.rs

```diff
@@ -3131,6 +3157,80 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		Ok(())
 	}
 
+	/// Attempts to forward an intercepted HTLC over the provided scid and with the provided
+	/// amount to forward. Should only be called in response to an [`HTLCIntercepted`] event.
+	///
+	/// Intercepted HTLCs can be useful for Lightning Service Providers (LSPs) to open a just-in-time
+	/// channel to a receiving node if the node lacks sufficient inbound liquidity.
+	///
+	/// To make use of intercepted HTLCs, use [`ChannelManager::get_intercept_scid`] to generate short
+	/// channel id(s) to put in the receiver's invoice route hints. These route hints will signal to
+	/// LDK to generate an [`HTLCIntercepted`] event when it receives the forwarded HTLC, and this
+	/// method or [`ChannelManager::fail_intercepted_htlc`] MUST be called in response to the event.
+	///
+	/// [`HTLCIntercepted`]: events::Event::HTLCIntercepted
+	pub fn forward_intercepted_htlc(&self, intercept_id: InterceptId, short_channel_id: u64, amt_to_forward_msat: u64) -> Result<(), APIError> {
+		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(&self.total_consistency_lock, &self.persistence_notifier);
+
+		let payment = self.pending_intercepted_htlcs.lock().unwrap().remove(&intercept_id)
+			.ok_or_else(|| APIError::APIMisuseError {
+				err: format!("Payment with InterceptId {:?} not found", intercept_id)
+			})?;
+
+		if self.short_to_chan_info.read().unwrap().get(&short_channel_id).is_none() {
+			return Err(APIError::APIMisuseError {
```

Comment:
A general question here, would we like to check here if the `short_channel_id` is within the `Intercept` namespace here, and add a specific error message if that's the case that informs the user that they shouldn't be sending in the `requested_next_hop_scid` from the `InterceptedHtlc` event to this function, and instead pass the `short_channel_id` of the channel they do want to forward over?

I think there's a risk that some users will misunderstand how they should be using this API function, and that could clarify what they're doing wrong. Though you likely know how our users behave better than me, so feel free to ignore this suggestion if you think it's dumb 😅

Reply:
I wouldn't be opposed to this, but I think the risk is somewhat ameliorated by this new parameter requirement: https://github.com/lightningdevkit/rust-lightning/pull/1835#discussion_r1019751007 

Also, in the future we're gonna expose `requested_next_hop: InterceptNextHop` (see this enum in `events.rs`) instead of `requested_next_hop_scid`. The enum will clearly delineate that it's a fake scid, which will hopefully help. Let me know your thoughts on this.

Reply:
Yeah that definitely addresses it :)! Any specific reason why we'd still like the API function to take the the `scid` + `node_id` instead of `node_id` + `channel_id` as https://github.com/lightningdevkit/rust-lightning/pull/1835#discussion_r1019751007 suggests? I think it kinda makes sense to use the `channel_id` instead as it's a bit more consistent with our how other API calls that refers to specific channels.

Reply:
Oh you're so right, I misread Matt's comment. Thanks, will fix!